### ISSUE TITLE
Remove the autopush crate

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-services/autopush-rs#524]
At some point we can remove the autopush crate entirely from the repo, probably earliest some time in Q1 '24.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-4036)

**Comment by data-sync-user** [comment]
➤ Philip Jenvey commented:

To fully cleanup the all artifacts when removing this crate:

  * Most of the dependencies solely used by the legacy autopush crate should solely have entries in its Cargo.toml, but there’s likely also some in the root Cargo.toml file. There’s a few different ways to detect these – there should be a pass done to remove them
  * This crate’s responsible for likely most if not all of our cargo audit ignores. So all the ignores in .cargo/audit.toml should be removed, then cargo audit should be ran again to see if any advisories still need to continue being ignored

**Comment by data-sync-user** [comment]
➤ Taddes Korris commented:

Just noting a few things:

  * Ran cargo machete to check on unused deps in parent/root Cargo.toml. Did a little reviewing and regex searches too. In the end, it was quite clean and most of the deps pertaining to the autopush crate were unique to its own `Cargo.toml`.
  * Ref to module in integration tests removed
  * A handful of audit messages were removed. Those that remain do pertain to other dependencies in the workspace. I renamed one that related to tungstenite that was tokio-tungstenite in the autopush crate. However, tungstenite in the parent crate is set to older ver (Note from Cargo.toml: Use older version of tungstenite to support legacy connection server.)
